gen/map.go

```diff
@@ -281,3 +281,144 @@ func (m *mapGenerator) equalsUnhashable(g Generator, spec *compile.MapSpec) (str
 
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
+
+func (m *mapGenerator) zapMarshaler(
+	g Generator,
+	spec compile.TypeSpec,
+	root *compile.MapSpec,
+	fieldValue string,
+) (string, error) {
+	name := zapperName(g, spec)
+	switch root.KeySpec.(type) {
+	case *compile.StringSpec:
```

Comment:
can we split these two cases out into two functions called from here? would help with readabilty a bit.